block/src/Pos/Block/Behavior.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE RankNTypes #-}
+module Pos.Block.Behavior
+       ( BlockBehavior (..)
+       , ForgeHeaderParams (..)
+       , HasBlockBehavior
+       , withBlockBehavior
+       , blockBehavior
+       ) where
+
+import           Universum
+
+import qualified Data.Aeson as A
+import           Data.Default (Default (..))
+import           Data.Reflection (Given (..), give, given)
+import           Serokell.Aeson.Options (defaultOptions)
+
+import           Pos.Util.Util (toAesonError)
+
+----------------------------------------------------------------------------
+-- Types for the behavior config
+----------------------------------------------------------------------------
+
+-- | Block settings (a part of the behavior config).
+--
+-- The syntax of this config section is as follows:
+--
+-- @
+-- block:
+--     forgeHeader: Normal | WrongLeader
+-- @
+data BlockBehavior = BlockBehavior
+    { -- | Block header forging settings
+      bbForgeHeader :: !ForgeHeaderParams
+    }
+    deriving (Eq, Show, Generic)
+
+data ForgeHeaderParams
+    = HeaderNormal      -- ^ Do not forge anything
+    | HeaderWrongLeader -- ^ Create blocks header with a wrong slot leader
+                        --    signature
```

Comment:
This comment is ambiguous. It can mean that we put wrong leader or wrong signature (or maybe both).